src/hotspot/cpu/x86/vm_version_x86.cpp

```diff
@@ -902,7 +932,11 @@ void VM_Version::get_processor_features() {
     }
   }
   if (UseAVX > use_avx_limit) {
-    warning("UseAVX=%d is not supported on this CPU, setting it to UseAVX=%d", (int) UseAVX, use_avx_limit);
+    if (UseSSE < 4) {
+      warning("UseAVX=%d requires UseSSE=4, setting it to UseAVX=0", (int) UseAVX);
+    } else {
+      warning("UseAVX=%d is not supported on this CPU, setting it to UseAVX=%d", (int) UseAVX, use_avx_limit);
+    }
     FLAG_SET_DEFAULT(UseAVX, use_avx_limit);
   } else if (UseAVX < 0) {
```

Comment:
`UseAVX < 0` check can also be cleaned up.

Reply:
Right, the flag range predicate check precedes this code, thus UseAVX < 0 is impossible and this code can be removed.